includes/admin/class-wc-payments-admin.php

```diff
@@ -315,6 +272,49 @@ public function add_payments_menu() {
 		}
 		global $submenu;
 
+		$this->admin_child_pages = [
```

Comment:
I considered this, but decided to keep it as-is.

> Currently, $this->admin_child_pages is populated in add_payments_menu and used in both add_payments_menu and maybe_redirect_from_payments_admin_child_pages

I'd just like to mention that this is not something that is getting introduced within this PR, but has been this way for a while, since the final `admin_child_pages` is only available after `add_payments_menu` runs. That's because the method adds and modifies pages.

> There’s a possibility (or maybe not) that $this->admin_child_pages could be empty in the redirect function if it hasn’t been populated yet, causing redirects to fail.

The idea of the method is to redirect away from a page, if the user does not access it. There is technically no page before the `admin_menu` 

> That said, since `maybe_redirect_from_payments_admin_child_pages` is hooked to the `admin_init` action, there’s a chance this change might reintroduce the issue that this PR was intended to solve. 

That's another +1 for keeping it as is: It requires `maybe_redirect_from_payments_admin_child_pages` to run late, and prevents it from meaningfully running before translations are properly loaded.

That's to say, I'm merging the PR with the current structure.